### DISCUSSION TITLE
Tandoor UI improvements

### DISCUSSION THREAD

**Discussion by maartenbrakkee** [TandoorRecipes/recipes#2817] [Ideas]
_As I couldn't find any related discussion, I'm creating this one._

The following is the first item of the [roadmap list](https://docs.tandoor.dev/#roadmap) on the docs site:
>Improve the UI! The Design is inconsistent and many pages work but don't look great. This needs to change.

To kickstart the conversation, it would be valuable to involve @vabene1111 in the discussion. Are there any preliminary ideas or strategies in mind to achieve the desired changes?

**Comment by smilerz** [comment]
Few of the primary maintainers have any UI experience - if you have skills and ideas on how to improve it then please put forth your ideas.  The only truly hard requirement is the front end must be kept separate from the backend.  

Some other random thoughts/needs:
- Vue needs upgraded to 3 (or maybe different framework?) before any major changes take place.
- Broader use of shared components.
- Front-end unit tests.
- Views that leverage cards should also support compact list/table view.
- Broader configurability to enable/disable features on System, Space, User level.

**Comment by maartenbrakkee** [comment]
While I haven't looked into the codebase yet, I'd suggest initiating the following strategic steps based on the information provided:

- [ ] Transition to Vue 3 as Vue 2 reaches its end of life on December 31st, 2023.
- [ ] Implement the Quasar UI library, known for its comprehensive documentation, robust support, and a rich set of components.
- [ ] Incorporate front-end unit tests for enhanced code reliability.

Unfortunately, my current schedule doesn't allow me to take up this amount of work. Therefore, I propose creating well-defined issues for the steps above and tag these issues as "help wanted" to encourage community contributions and collaboration.

**Comment by vabene1111** [comment]
thanks for all the well ment ideas and suggestions, but I feel like that is quite a lot of work for the hope that somebody might pick things up, espcically since many of the thigns are already documented in issues.

Things will progress but for many things not progressing there is a reason, explaing everything takes to much time, if anyone wants to help they can
